internal/clients/sources/sources_interface.go

```diff
@@ -0,0 +1,12 @@
+package sources
+
+import (
+	"context"
+)
+
+var GetSourcesImpl func(ctx context.Context) (SourcesIntegration, error)
+
+type SourcesIntegration interface {
+	GetById(ctx context.Context, id string) (*Source, error)
+	List(ctx context.Context) (*[]Source, error)
```

Comment:
I'd have `ListSourcesWithResponse` and `ShowSourceWithResponse` methods here, that would copy the method interface from the real thing. Wrapping can be useful, but I'd love to stick with the real thing until we hit the need for wrapping..